ui-speedspacechart/src/components/helpers/drawElements/speedLimits.ts

```diff
@@ -0,0 +1,102 @@
+import type { DrawFunctionParams } from '../../../types/chartTypes';
+import { ERROR_30, ERROR_60, MARGINS, WARNING_30 } from '../../const';
+import {
+  clearCanvas,
+  maxPositionValue,
+  maxSpeedValue,
+  metersToKilometers,
+  positionToPosX,
+} from '../../utils';
+
+const { CURVE_MARGIN_TOP, MARGIN_RIGHT, MARGIN_LEFT, MARGIN_BOTTOM, MARGIN_TOP } = MARGINS;
+const GRADIENT_HEIGHT = 16;
+const TRAIN_LENGTH_MARKER_HEIGHT = 8;
+
+export const drawSpeedLimits = ({ ctx, width, height, store }: DrawFunctionParams) => {
+  const { mrsp, trainLength, ratioX, leftOffset } = store;
+
+  if (mrsp === undefined) {
+    return;
+  }
+
+  clearCanvas(ctx, width, height);
+
+  ctx.save();
+  ctx.translate(leftOffset, 0);
+
+  const realHeight = height - MARGIN_BOTTOM - MARGIN_TOP;
+  const maxSpeed = maxSpeedValue(store);
+  const maxPosition = maxPositionValue(store);
+  // Add the last boundary to the boundaries array
+  mrsp.boundaries.push(maxPosition);
+
+  ctx.lineCap = 'round';
+
+  let previousBoundaryX = positionToPosX(0, maxPosition, width, ratioX);
+  let previousSpeedY: number | null = null;
+  for (let i = 0; i < mrsp.values.length; i++) {
+    const { speed, isTemporary } = mrsp.values[i];
+
+    const currentBoundaryX = positionToPosX(mrsp.boundaries[i], maxPosition, width, ratioX);
+
+    const speedY = realHeight - (speed / maxSpeed) * (realHeight - CURVE_MARGIN_TOP) + MARGIN_TOP;
+    // Draw vertical line
```

Comment:
As we have 2 vertical lines, maybe we could be more precize

```suggestion
    // Draw vertical line joining 2 speed limits
```